app/ui/src/Components.elm

```diff
@@ -0,0 +1,33 @@
+module Components exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onInput, onSubmit)
+import Model exposing (Output(..))
+
+
+layout : List (Html msg) -> Html msg
+layout children =
+    div [] children
+
+
+query : (String -> msg) -> msg -> Html msg
+query queryChange submit =
+    Html.form [ onSubmit submit ]
+        [ label [ for "query" ] [ text "Your query: " ]
+        , input [ id "query", onInput queryChange ] []
+        , button [ type_ "submit", onClick submit ] [ text "Update" ]
+        ]
+
+
+output : Output -> Html msg
```

Comment:
The List type provides all this type does so you can simplify it to this:

```elm
type alias Output
    = List String

output : Output -> Html msg
output output =
    case output of
        [message] ->
            p [] [ text message ]

        messages ->
            div []
                [ p [] [ text "Server messages:" ]
                , ul [] (List.map (\l -> li [] [ text l ]) messages)
                ]
```

Output isn't a very clear name I think